lib/utils/SystemInterface.cpp

```diff
@@ -0,0 +1,20 @@
+#include "SystemInterface.hpp"
+
+#include <string>   // for string, operator+
+#include <unistd.h> // for access, F_OK
+
+const std::string get_user_name() {
+
+    std::string user(256, '\0');
+    user = (getlogin_r(&user[0], 256) == 0) ? user.c_str() : "unknown";
+
+    return user;
+}
+
+const std::string get_host_name() {
```

Comment:
These are great. Although I think *hostname* and *username* would be preferred, e.g. https://english.stackexchange.com/questions/100858/hostname-or-host-name